### ISSUE TITLE
Login page does not open sometimes on OS X Monterrey

### ISSUE THREAD

**Issue by shuw** [auth0/Auth0.swift#621]
### Describe the problem

On OS X Monterrey (confirmed with, 12.0.1), the browser sign-in page may occasionally not open after clicking "continue". This issue was seen with Safari set as the default browser.

Debugging the issue, the callback from Auth0 does not occur as well. I suspect this could be an issue with Auth0's usage of ASWebAuthenticationSession (which owns that the dialog and the invocation of the browser. There is a report of a related issue on Apple's developer forum: https://developer.apple.com/forums/thread/694465.

### What was the expected behavior?
The browser sign-in page is always opened after clicking "continue" in the prompt to sign-in.

### Reproduction
1. Start in a signed-out state
2. Begin the Auth0 authentication flow
3. See the dialog "AppX" wants to use "auth0.com" to sign in.
4. Click continue

**Can the behavior be reproduced using the [Auth0.swift sample app](https://github.com/auth0-samples/auth0-ios-swift-sample)?**
N/A - issue is on MacOS, not iOS

### Environment
- **Version of `Auth0.swift` used: 1.38.0**
- **Version of MacOS: 12.0.1**
- **Version of Xcode: 13.2**

**Comment by Widcket** [comment]
Hi @shuw, thanks for raising this. Unfortunately, this seems like an `ASWebAuthenticationSession` bug on Monterey, and not an Auth0.swift bug. I'll be closing this issue, as it is not actionable by us.

**Comment by shuw** [comment]
@Widcket is there any possibility that Auth0 could support an alternate approach than using ASWebAuthenticationSession on MacOS?

**Comment by Widcket** [comment]
Unlike on iOS, `ASWebAuthenticationSession` is the only way to perform web-based authentication on macOS that does not involve embedding a web-server to listen for the callback.

**Comment by jpr5** [comment]
@Widcket Is this the first time this issue has actually been reported?  We've been encountering it _regularly_, and I just have to think that if there's any meaningful macOS+Auth0 @ Monterey install base, they're 100% guaranteed to run into this.  

I get the WONTFIX/CANTFIX diagnosis, but shouldn't this rate something more support-wise for the users, given the likelihood of them all experiencing it + the likelihood of the first point of support/blame being Auth0?  Like a KB article or FAQ entry or something?  Or maybe you folks have a contact at Apple that can elevate attention to it?

**Comment by Widcket** [comment]
Hi @jpr5, this is the first time this issue has been reported to us. I'll pin this issue for greater visibility.

**Comment by jpr5** [comment]
Thanks, appreciate it.

**Comment by sincaleb** [comment]
I hate to just drop in and say "Me Too", but a similar issue happens to us very frequently as well. We are using an ephemeral session so we don't get the prompt, but sometimes the login window just never appears. If we have our users quit safari and quit our catalyst app it functions normally, but its a pretty frequent occurrence and we have been getting a lot of complaints about it. Thanks!

**Comment by Kardioversion** [comment]
Could you all put this in the documentation as a "known issue"? Or at least do not close issues that are unresolved?

I scanned this repo a few hours ago and there were zero open issues, so I moved on to other possible causes of this bug. I understand Auth0 is not causing the bug but if your SDK does not work on a major platform (macOS), then I consider that an open issue, even if internally at Auth0 it is not a "we need to fix this issue by writing code" situation.

**Comment by shuw** [comment]
For others experiencing this issue and looking for a workaround, my solution was to move off of Auth0's SDK for doing the authentication on OSX.

Instead, I used a library called OAuthSwift to perform the authentication with a regular browser window. And there are 2 ways to receive the authentication token back:
* Apple Custom URL Scheme (easiest)
* Build a local HTTP server (I used Swifter) to receive the token back. This approach is more complicated but avoids an extra dialog that comes up when a custom URL scheme is invoked.
